src/components/plots/abstractPlot.ts

```diff
@@ -429,6 +429,46 @@ export module Plot {
 
       return allSelections;
     }
+
+    /**
+     * Gets the selections under the given pixel position (if [xValOrExtent]
+     * and [yValOrExtent] are {number}s), under a given line (if only one
+     * of [xValOrExtent] or [yValOrExtent] are {Extent}s) or are under a
+     * 2D area (if [xValOrExtent] and [yValOrExtent] are both {Extent}s).
+     *
+     * @param {number | Extent} xValOrExtent The pixel x position, or range of x values.
+     * @param {number | Extent} yValOrExtent The pixel y position, or range of y values.
+     * @returns {D3.Selection} The selections within under the given bounds
+     */
+    public getSelections(xValOrExtent: number | Extent, yValOrExtent: number | Extent, tolerance = 0.5): D3.Selection {
+      var xExtent: Extent = AbstractPlot._parseExtent(xValOrExtent);
+      var yExtent: Extent = AbstractPlot._parseExtent(yValOrExtent);
+
+      var selections: EventTarget[] = [];
+      this._datasetKeysInOrder.forEach((key: string) => {
+        var drawer = this._key2PlotDatasetKey.get(key).drawer;
+        drawer._getRenderArea().selectAll(drawer._getSelector()).each(function () {
+          var selection = d3.select(this);
+          // Check if extent covers the selection
+          if (drawer._isSelectionInBounds(selection, xExtent, yExtent, tolerance)) {
+            selections.push(this);
+          }
+        });
+      });
+
+      return d3.selectAll(selections);
+    }
+
+    private static _parseExtent(input: any): Extent {
```

Comment:
Might want to make this `protected`, since I can see subclass implementations wanting to use it.